packages/kbn-eslint-plugin-imports/src/get_import_resolver.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ImportResolver } from '@kbn/import-resolver';
+import { REPO_ROOT } from '@kbn/utils';
+import { Rule } from 'eslint';
+import { RUNNING_IN_EDITOR } from './helpers/running_in_editor';
+
+let importResolverCache: ImportResolver | undefined;
+
+/**
+ * Create a request resolver for ESLint, requires a PluginPackageResolver from @kbn/bazel-packages which will
+ * be created and cached on contextServices automatically.
+ *
+ * All import requests in the repository should return a result, if they don't it's a bug
+ * which should be caught by the `@kbn/import/no_unresolved` rule, which should never be disabled. If you need help
+ * adding support for an import style please reach out to operations.
+ */
+export function getImportResolver(context: Rule.RuleContext): ImportResolver {
+  if (RUNNING_IN_EDITOR) {
+    return (context.parserServices.kibanaImportResolver ||= ImportResolver.create(REPO_ROOT));
```

Comment:
The goal is to avoid rerunning discovery when the context hasn't changed and maintaining the cache there so ESLint can control when the recreate the cache.

Reply:
Is there any reason why then we don't use the same logic even when not running in an editor? Is the context not valid for those cases and do we need to hold it on another variable?

Reply:
The context is recreated for each file, so when running on the CLI it would re-run discovery for every file. In the editor this just prevents multiple rules from re-triggering discovery when linting a single file.